### PR TITLE
Split media manager

### PR DESCRIPTION
**Pull request by kanavin**
Alright, I just went ahead and did it (NYE notwithstanding). Happy New Year!

### CODE REVIEW

File: wfd/common/message_handler.h [hunk 1]
```diff
@@ -79,6 +78,14 @@ class MessageHandler {
     assert(observer_);
   }
 
+  inline SourceMediaManager* GetSourceMediaManager() {
```
Comment (pozdnyakov) [inline wfd/common/message_handler.h]: s/GetSourceMediaManager/ToSourceMediaManager

Comment (pozdnyakov) [inline wfd/common/message_handler.h]: make it a free function (located in the same place where SourceMediaManager is declared)

File: wfd/common/message_handler.h [hunk 2]
```diff
@@ -79,6 +78,14 @@ class MessageHandler {
...
+    return static_cast<SourceMediaManager*>(manager_);
+  }
+
+  inline SinkMediaManager* GetSinkMediaManager() {
```
Comment (pozdnyakov) [inline wfd/common/message_handler.h]: ditto

File: wfd/common/message_handler.h (hunk 1 above)
Comment (kanavin) [inline wfd/common/message_handler.h]: Why?

File: sink/gst_sink_media_manager.cpp [hunk 3]
```diff
@@ -19,52 +19,44 @@
  * 02110-1301 USA
  */
 
-#include "sink_media_manager.h"
+#include "gst_sink_media_manager.h"
 
-SinkMediaManager::SinkMediaManager(const std::string& hostname)
+GstSinkMediaManager::GstSinkMediaManager(const std::string& hostname)
```
Comment (alexshalamov) [inline sink/gst_sink_media_manager.cpp]: Should be in separate commit.
One commit for interface spit
One commit for renaming

File: wfd/common/message_handler.h (hunk 1 above)
Comment (pozdnyakov) [inline wfd/common/message_handler.h]: this is a conversion function -- not a getter. ToSourceMediaManager(MediaManager* from)

**Comment by kanavin** [comment]
All fixed. How does it look now?

File: wfd/public/media_manager.h [hunk 4]
```diff
@@ -31,15 +31,35 @@ class MediaManager {
...
   virtual std::string PresentationUrl() const = 0;
   virtual void SetSession(const std::string& session) = 0;
   virtual std::string Session() const = 0;
+
+};
+
+class SourceMediaManager : public MediaManager {
+ public:
+  virtual void SetSinkRtpPorts(int port1, int port2) = 0;
+  virtual std::pair<int,int> SinkRtpPorts() const = 0;
```
Comment (pozdnyakov) [inline wfd/public/media_manager.h]: SinkRtpPorts() is present in both SourceMediaManager and SinkMediaManager, does it make sense to keep it in the parent interface?

**Comment by pozdnyakov** [comment]
lgtm

File: wfd/public/media_manager.h (hunk 4 above)
Comment (kanavin) [inline wfd/public/media_manager.h]: I'd say it's better to rename SinkRtpPorts() in SinkMediaManager to ListeningRtpPorts(). I can add that as a separate commit.

**Comment by kanavin** [comment]
I added the renaming patch just now - all good?
